packages/webapp/webapp_server.js

```diff
@@ -815,7 +816,55 @@ var runWebAppServer = function () {
     var localPort = WebAppInternals.parsePort(process.env.PORT) || 0;
     var host = process.env.BIND_IP;
     var localIp = host || '0.0.0.0';
-    httpServer.listen(localPort, localIp, Meteor.bindEnvironment(function() {
+
+    if (typeof localPort == "number")
+    {
+      var listenOptions = { port: localPort, host: host };
+    }
+    else
+    {
+      var socketPath = localPort;
+      var listenOptions = { path: socketPath };
+
+      httpServer.on('error', Meteor.bindEnvironment(function(e) {
+
+        if (e.code == 'EADDRINUSE') {
+
+	  var clientSocket = new net.Socket();
+
+          clientSocket.on('error', Meteor.bindEnvironment(function(e) {
+
+            if (e.code == 'ECONNREFUSED') {
+
+              console.log("Deleting stale socket file");
+              fs.unlinkSync(socketPath);
+
+              httpServer.listen(listenOptions, Meteor.bindEnvironment(function() {
```

Comment:
It looks like this `httpServer.listen` block of code is repeated a few lines down, and is the exact same. Could this be extracted into a common function to clean the code up a bit?